features/unsupported/USBHost/USBHostMSD/USBHostMSD.cpp

```diff
@@ -323,44 +337,76 @@ int USBHostMSD::disk_initialize() {
     return readCapacity();
 }
 
-int USBHostMSD::disk_write(const uint8_t* buffer, uint32_t block_number, uint32_t count) {
-    USB_DBG("FILESYSTEM: write block: %lld, count: %d", block_number, count);
+int USBHostMSD::program(const void *buffer, bd_addr_t addr, bd_size_t size)
+{
+    uint32_t block_number, count;
+    uint8_t *buf = (uint8_t *)buffer;
     if (!disk_init) {
-        disk_initialize();
+        init();
     }
     if (!disk_init)
         return -1;
+    block_number =  addr / blockSize;
+    count = size /blockSize;
+
     for (uint32_t b = block_number; b < block_number + count; b++) {
-        if (dataTransfer((uint8_t*)buffer, b, 1, HOST_TO_DEVICE))
+        if (dataTransfer(buf, b, 1, HOST_TO_DEVICE))
             return -1;
-        buffer += 512;
+        buf += blockSize;
     }
     return 0;
 }
 
-int USBHostMSD::disk_read(uint8_t* buffer, uint32_t block_number, uint32_t count) {
-    USB_DBG("FILESYSTEM: read block: %lld, count: %d", block_number, count);
+int USBHostMSD::read(void *buffer, bd_addr_t addr, bd_size_t size)
+{
+    uint32_t block_number, count;
+    uint8_t *buf = (uint8_t *)buffer;
     if (!disk_init) {
-        disk_initialize();
+        init();
     }
     if (!disk_init)
         return -1;
+    block_number =  addr / blockSize;
+    count = size / blockSize;
+
     for (uint32_t b = block_number; b < block_number + count; b++) {
-        if (dataTransfer((uint8_t*)buffer, b, 1, DEVICE_TO_HOST))
+        if (dataTransfer(buf, b, 1, DEVICE_TO_HOST))
             return -1;
-        buffer += 512;
+        buf += blockSize;
     }
     return 0;
 }
 
-uint32_t USBHostMSD::disk_sectors() {
-    USB_DBG("FILESYSTEM: sectors");
-    if (!disk_init) {
-        disk_initialize();
-    }
+int USBHostMSD::erase(bd_addr_t addr, bd_size_t size)
+{
+    return 0;
+}
+
+bd_size_t USBHostMSD::get_read_size() const
+{
     if (!disk_init)
-        return 0;
-    return blockCount;
+        return -1;
+    return (bd_size_t)blockSize;
+}
+
+bd_size_t USBHostMSD::get_program_size() const
+{
+    if (!disk_init)
+        return -1;
+    return  (bd_size_t)blockSize;
+}
+bd_size_t USBHostMSD::get_erase_size() const
+{
+    if (!disk_init)
+        return -1;
+    return  (bd_size_t)blockSize;
 }
 
+bd_size_t USBHostMSD::size() const
+{
+    USB_DBG("FILESYSTEM: size ");
+    if (!disk_init)
```

Comment:
if we are changing these lines, we shall update the style in here as it was done for the rest of the code?